src/cmd/services/m3nsch_server/main/main.go

```diff
@@ -97,9 +97,7 @@ func newSessionFn(conf config.Configuration, iopts instrument.Options) m3nsch.Ne
 		if zone != svc.Zone {
 			return nil, fmt.Errorf("request zone %s did not match config zone %s", zone, svc.Zone)
 		}
-		if env != svc.Env {
-			return nil, fmt.Errorf("request env %s did not match config zone %s", env, svc.Env)
-		}
+		svc.Env = env
```

Comment:
Don't have context for this change - why override?

Reply:
Ah I should've left a comment. AFAICT this was causing me to have to have a separate agent config for each environment I wanted to load test, since if my `env` flag didn't match the YAML it bailed.

What I wanted was to be able to have a bunch of agent deployments share one config with the etcd cluster and override env per-tier. Or be able to start a benchmark with a set of agents, stop them, and then resume them against a different env.

Reply:
Edit: for some reason I didn't relax this check for zone, I forget but can change that if we want the behavior more aligned

Reply:
Makes sense. Maybe cleanup zone as you suggested too